featurebyte/models/request_input.py

```diff
@@ -148,9 +157,44 @@ async def materialize(
                 columns = self.columns
             query_expr = select_and_rename_columns(query_expr, columns, self.columns_rename_mapping)
 
+        # apply time range filter if sample_from_timestamp and sample_to_timestamp are provided
+        time_range_conditions: list[expressions.Expression] = []
+        adapter = get_sql_adapter(session.get_source_info())
+        if sample_from_timestamp is not None:
+            time_range_conditions.append(
+                expressions.GTE(
+                    this=adapter.normalize_timestamp_before_comparison(
+                        quoted_identifier(SpecialColumnName.POINT_IN_TIME)
```

Comment:
Can we assume that `POINT_IN_TIME` always exist in `query_expr`?

Reply:
I think `POINT_IN_TIME` is a required column for any observation table after column renaming is complete. Do you see any case that that is not the case?

Reply:
I don't see any other case since this is only used for materializing observation tables so a valid one should have `POINT_IN_TIME`. Just need to make sure that the validation is performed before `query_expr` is executed (I think that is the case).

Reply:
The validation for columns are done at the time of payload creation so it should be before this is called in the worker task